synapse/config/_base.py

```diff
@@ -242,12 +242,11 @@ def read_templates(
         env = jinja2.Environment(loader=loader, autoescape=autoescape)
 
         # Update the environment with our custom filters
-        env.filters.update(
-            {
-                "format_ts": _format_ts_filter,
-                "mxc_to_http": _create_mxc_to_http_filter(self.public_baseurl),
-            }
-        )
+        env.filters.update({"format_ts": _format_ts_filter})
+        if self.public_baseurl:
+            env.filters.update(
+                {"mxc_to_http": _create_mxc_to_http_filter(self.public_baseurl)}
+            )
```

Comment:
Per the conversation in #synapse-dev:matrix.org it seems that `public_baseurl` is not required to be set. I believe any feature that expects to use this template filter already asserts that.

Reply:
Seems so!